database/neo4j/neo4j.go

```diff
@@ -108,28 +107,29 @@ func (n *Neo4j) Run(migration io.Reader) error {
 		return err
 	}
 	defer func() {
-		err := session.Close()
-		if err != nil {
-			log.Printf("error: %s", err)
+		if cerr := session.Close(); cerr != nil {
+			err = cerr
 		}
 	}()
 
 	result, err := session.Run(string(body[:]), nil)
 	if err != nil {
 		return err
 	}
-	return result.Err()
+	if result.Err() != nil {
+		err = result.Err()
```

Comment:
It's cleaner to do:
```golang
if e := result.Err(); e != nil {
    return e
}
return nil
```
e.g. there are 2 main concerns:
1. Don't call `Err()` multiple times in case the value changes between calls (shouldn't happen, but better safe than sorry)
1. Explicitly `return nil` on success (the defer will override any existing value for `err`)